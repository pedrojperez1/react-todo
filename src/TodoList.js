import React, {useState} from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";

const TodoList = () => {
    const [todos, setTodos] = useState([]);
    const addTodo = (newTodo) => {
        setTodos([...todos, newTodo]);
    };
    const removeTodo = (id) => {
        setTodos(todos.filter(todo => todo.id !== id));
    }
    return (
        <div className="TodoList">
            <NewTodoForm addTodo={addTodo}/>
            <ul>
                {todos.map(todo => <Todo text={todo.todo} id={todo.id} removeTodo={removeTodo}/>)}
            </ul>
        </div>
    )
};

export default TodoList;