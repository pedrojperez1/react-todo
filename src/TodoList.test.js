import { render, fireEvent } from '@testing-library/react';
import TodoList from './TodoList';

it("should render without going up in smoke", () => {
    render(<TodoList />);
  });
  
it("should render the same every time", () => {
  const {asFragment} = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should render a new todo when form is submitted and delete it when X is pressed", () => {
  const { getByPlaceholderText, getByText, queryAllByText } = render(<TodoList />);
  const todoInput = getByPlaceholderText("New Todo");
  const submitBtn = getByText("Add");
  fireEvent.change(todoInput, {target: {value: "my test todo"}});
  fireEvent.click(submitBtn);
  
  const newTodo = getByText("my test todo");
  expect(newTodo).toBeInTheDocument(); //first check
  fireEvent.click(getByText("X"));
  expect(newTodo).not.toBeInTheDocument(); //disappearance check
});
  