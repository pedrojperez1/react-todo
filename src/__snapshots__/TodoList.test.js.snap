// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the same every time 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <div>
      <form>
        <input
          name="todo"
          placeholder="New Todo"
          type="text"
          value=""
        />
        <button>
          Add
        </button>
      </form>
    </div>
    <ul />
  </div>
</DocumentFragment>
`;
